### PR TITLE
Add instructions how to setup the postgres user and clarify the final step

### PR DESCRIPTION
**Pull request by rubo77**
Signed-off-by: Ruben Barkow <github@r.z11.de>

### CODE REVIEW

**Comment by rubo77** [comment]
finding out, how to create the user and password for the postgresql database took us 2 hours to find out. so with this small hint, you help users that are not used to Postgresql a lot.

**Comment by rubo77** [comment]
I rebased and enhanced the instructions.

File: docs/postgres.rst [hunk 1]
```diff
@@ -6,6 +6,17 @@ Postgres version 9.4 or later is known to work.
 Set up database
 ===============
 
+Create a user ``synapse_user`` with::
+
+ su - postgres
```
Comment (richvdh) [inline docs/postgres.rst]: this is installation-specific. Not all distributions will have a unix user called `postgres`.

File: docs/postgres.rst [hunk 2]
```diff
@@ -46,8 +57,7 @@ As with Debian/Ubuntu, postgres support depends on the postgres python connector
 Synapse config
 ==============
 
-When you are ready to start using PostgreSQL, add the following line to your
-config file::
+When you are ready to start using PostgreSQL, edit the `database` section in your config file so they match the following lines::
```
Comment (richvdh) [inline docs/postgres.rst]: please wrap at 80 chars

File: docs/postgres.rst [hunk 3]
```diff
@@ -117,6 +127,6 @@ run::
         --postgres-config homeserver-postgres.yaml
 
 Once that has completed, change the synapse config to point at the PostgreSQL
-database configuration file ``homeserver-postgres.yaml`` (i.e. rename it to 
-``homeserver.yaml``) and restart synapse. Synapse should now be running against
-PostgreSQL.
+database configuration file ``homeserver-postgres.yaml`` (for an example
```
Comment (richvdh) [inline docs/postgres.rst]: I think we should continue to tell people to rename it.

Comment (richvdh) [reply docs/postgres.rst]: why should it be called `homeserver-postgres.yaml` rather than just `homeserver.yaml`?

Comment (rubo77) [reply docs/postgres.rst]: I thought, the name is irrelevant. Just this instruction here should use only one name as example

Comment (rubo77) [reply docs/postgres.rst]: I think the difficult part here is, that you have to create that file yourself with the example above. It is irrelevant how you name it

Comment (richvdh) [reply docs/postgres.rst]: You're telling people to call it `homeserver-postgres.yaml`.

And while we're here, the user should already have written the file, so pointing them to example content now is unhelpful.

**Comment by rubo77** [comment]
I didn't want to argue either. I just struggled to understand the meaning of the original wording.

I hope now it is clearer, what there is to do.

File: docs/postgres.rst [hunk 4]
```diff
@@ -6,7 +6,19 @@ Postgres version 9.4 or later is known to work.
 Set up database
 ===============
 
-The PostgreSQL database used *must* have the correct encoding set, otherwise
+Assuming your PostgreSQL database user is called ``postgres``, create a user
```
Comment (rubo77) [inline docs/postgres.rst]: I am not sure if this is the most elegant solution to create the user, I have no system at the moment to try out a shorter solution. 

feel free to edit this section if there is a shorter solution.

File: docs/postgres.rst [hunk 5]
```diff
@@ -46,8 +58,8 @@ As with Debian/Ubuntu, postgres support depends on the postgres python connector
 Synapse config
 ==============
 
-When you are ready to start using PostgreSQL, add the following line to your
-config file::
+When you are ready to start using PostgreSQL, edit the `database` section in
```
Comment (richvdh) [inline docs/postgres.rst]: RST uses double-backticks for preformatted text

File: docs/postgres.rst [hunk 6]
```diff
@@ -6,7 +6,19 @@ Postgres version 9.4 or later is known to work.
 Set up database
 ===============
 
-The PostgreSQL database used *must* have the correct encoding set, otherwise
+Assuming your PostgreSQL database user is called ``postgres``, create a user
+``synapse_user`` with::
+
+ su - postgres
+ createuser --pwprompt synapse_user
+ su - synapse_user
```
Comment (richvdh) [inline docs/postgres.rst]: this bit is now unnecessary

Comment (richvdh) [inline docs/postgres.rst]: this bit (up to line 19) is now unnecessary: the pwprompt allows you to enter a password

File: docs/postgres.rst [hunk 7]
```diff
@@ -117,6 +132,11 @@ run::
...
 
 Once that has completed, change the synapse config to point at the PostgreSQL
-database configuration file ``homeserver-postgres.yaml`` (i.e. rename it to 
-``homeserver.yaml``) and restart synapse. Synapse should now be running against
-PostgreSQL.
+database configuration file ``homeserver-postgres.yaml``:
+
+    ./synctl stop
+    cp homeserver.yaml homeserver-old-sqlite.yaml 
+    cp homeserver-postgres.yaml homeserver.yaml
```
Comment (richvdh) [inline docs/postgres.rst]: s/cp/mv/

File: docs/postgres.rst [hunk 8]
```diff
@@ -46,8 +58,8 @@ As with Debian/Ubuntu, postgres support depends on the postgres python connector
 Synapse config
 ==============
 
-When you are ready to start using PostgreSQL, add the following line to your
-config file::
+When you are ready to start using PostgreSQL, edit the `database` section in
+your config file so they match the following lines::
```
Comment (richvdh) [inline docs/postgres.rst]: "to match" sounds more natural than "so they match" here

File: docs/postgres.rst (hunk 6 above)
Comment (rubo77) [reply docs/postgres.rst]: this is outdated

Comment (rubo77) [reply docs/postgres.rst]: this is outdated

Comment (richvdh) [reply docs/postgres.rst]: ? it's still showing up in the diff at https://github.com/matrix-org/synapse/pull/1756/files

Comment (rubo77) [reply docs/postgres.rst]: Ok,so what lines have to be removed?

Comment (rubo77) [reply docs/postgres.rst]: done

**Pull request merged**